packages/hub/services/scheduled-payments/fetcher.ts

```diff
@@ -19,44 +21,74 @@ export default class ScheduledPaymentsFetcherService {
     let prisma = await this.prismaManager.getClient();
     let nowUtc = convertDateToUTC(new Date());
 
-    return prisma.scheduledPayment.findMany({
-      where: {
-        canceledAt: null,
-        creationBlockNumber: {
-          gt: 0,
-        },
-        payAt: {
-          gt: startOfDay(subDays(nowUtc, this.validForDays)),
-          lte: nowUtc,
-        },
-        OR: [
-          {
-            AND: {
-              recurringDayOfMonth: null,
-              scheduledPaymentAttempts: {
-                none: {
-                  OR: [{ status: 'succeeded' }, { status: 'inProgress' }],
-                },
-              },
-            },
-          },
-          {
-            AND: {
-              recurringDayOfMonth: {
-                gt: 0,
-              },
-              recurringUntil: {
-                gte: nowUtc,
-              },
-              scheduledPaymentAttempts: {
-                none: {
-                  status: 'inProgress',
-                },
-              },
-            },
-          },
-        ],
-      },
-    });
+    let results: [{ id: string; started_at: Date }] = await prisma.$queryRaw`SELECT 
+        "scheduled_payments"."id" AS "id",
+        "last_failed_payment_attempts"."started_at"
+      FROM 
+        "scheduled_payments"
+      LEFT JOIN (SELECT "scheduled_payment_attempts"."scheduled_payment_id",
+                  MAX("scheduled_payment_attempts"."started_at") AS "started_at"
+            FROM "scheduled_payment_attempts"
+            WHERE "scheduled_payment_attempts"."status" = 'failed'
+            GROUP BY "scheduled_payment_id") AS "last_failed_payment_attempts"
+      ON "last_failed_payment_attempts"."scheduled_payment_id" = "scheduled_payments"."id"
+      WHERE 
+        (
+          "scheduled_payments"."canceled_at" IS NULL 
+          AND "scheduled_payments"."creation_block_number" > 0 
+          AND "scheduled_payments"."pay_at" > ${startOfDay(subDays(nowUtc, this.validForDays))}
+          AND "scheduled_payments"."pay_at" <= ${nowUtc}
+          AND (
+            (
+              "scheduled_payments"."recurring_day_of_month" IS NULL 
+              AND (
+                "scheduled_payments"."id"
+              ) NOT IN (
+                SELECT 
+                  "t0"."id" 
```

Comment:
Is this SQL generated by a script? It doesn't look written by hand. `t0`, `j0` etc sound too cryptic, let's use human readable code. Also you can drop the "s, for example.  `"scheduled_payments"."recurring_day_of_month"` can be simplified to `scheduled_payments.recurring_day_of_month`

Reply:
Some of them I got from the previous prisma query. I will rename it.

Reply:
updated